Rakefile

```diff
@@ -65,24 +52,25 @@ def convert_ttml(source, format)
   end
 end
 
-sup_source = -> (filename) do
-  extname =
-    case filename
-    when /\.sup$/; /\.sup$/
-    when /\.ja\.sub$/; /\.ja\.sub$/
+def text_subtitle_sources(lang, extensions)
+  -> (filename) do
+    candidates = extensions.map do |ext|
+      filename.gsub(/\.#{lang}\..+/, ".#{ext}.ttml")
     end
-  source = filename.gsub(extname, ".日本語.ttml.zip")
-  break source if File.exist?(source)
-
-  glob = filename.gsub(extname, ".*.ttml.zip")
-  Dir.glob(glob).first || source
+    candidates.find do |source|
+      File.exist?(source)
+    end || candidates.first
+  end
 end
 
-rule '.sup' => sup_source do |t|
-  convert_ttml(t.source, "Blu-raysup") do |dir|
-    Dir.glob("#{dir}/*.sup").each do |filename|
-      FileUtils.cp filename, t.name
+def image_subtitle_sources(lang, extensions)
```

Comment:
this method is very similar with `text_subtitle_sources`.
would be better to sprout method.

Reply:
refactor it